src/main/java/io/nats/client/Options.java

```diff
@@ -2465,24 +2464,24 @@ public CharBuffer buildProtocolConnectOptionsString(String serverURI, boolean in
             }
 
             if (uriUser != null) {
-                appendOption(connectString, Options.OPTION_USER, uriUser, true, true);
+                appendOption(connectString, Options.OPTION_USER, jsonEncode(uriUser), true, true);

```

Comment:
json encode since this is json, and it's a string that is provided by the user. I said this to say that I didn't encode some strings in this function since I know that they are already json safe, like the jwt which is base64 encoded.